modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFolderLocalServiceTest.java

```diff
@@ -0,0 +1,326 @@
+import com.liferay.portal.kernel.test.util.CompanyTestUtil;
+import com.liferay.portal.kernel.test.util.RandomTestUtil;
+import com.liferay.portal.kernel.test.util.TestPropsValues;
+import com.liferay.portal.kernel.test.util.UserTestUtil;
+import com.liferay.portal.kernel.util.LocaleUtil;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.vulcan.util.LocalizedMapUtil;
+
+import java.util.Locale;
+import java.util.Map;
+
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Murilo Stodolni
+ */
+@RunWith(Arquillian.class)
+public class ObjectFolderLocalServiceTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new LiferayIntegrationTestRule();
+
+	@BeforeClass
+	public static void setUpClass() throws Exception {
+		_uncategorizedObjectFolder = _objectFolderLocalService.getObjectFolder(
+			TestPropsValues.getCompanyId(),
+			ObjectFolderConstants.NAME_UNCATEGORIZED);
+	}
+
+	@Test
+	public void testAddObjectFolder() throws Exception {
+		AssertUtils.assertFailure(
+			ObjectFolderLabelException.class,
+			"Label is null for locale " + LocaleUtil.US.getDisplayName(),
+			() -> _objectFolderLocalService.addObjectFolder(
+				RandomTestUtil.randomString(), TestPropsValues.getUserId(),
+				null, RandomTestUtil.randomString()));
+
+		AssertUtils.assertFailure(
+			ObjectFolderNameException.MustNotBeNull.class, "Name is null",
+			() -> _objectFolderLocalService.addObjectFolder(
+				RandomTestUtil.randomString(), TestPropsValues.getUserId(),
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				null));
+
+		AssertUtils.assertFailure(
+			ObjectFolderNameException.MustBeLessThan41Characters.class,
+			"Name must be less than 41 characters",
+			() -> _objectFolderLocalService.addObjectFolder(
+				RandomTestUtil.randomString(), TestPropsValues.getUserId(),
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				RandomTestUtil.randomString(42)));
+
+		AssertUtils.assertFailure(
+			ObjectFolderNameException.MustOnlyContainLettersAndDigits.class,
+			"Name must only contain letters and digits",
+			() -> _objectFolderLocalService.addObjectFolder(
+				RandomTestUtil.randomString(), TestPropsValues.getUserId(),
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				"Abl e"));
+
+		AssertUtils.assertFailure(
+			ObjectFolderNameException.MustOnlyContainLettersAndDigits.class,
+			"Name must only contain letters and digits",
+			() -> _objectFolderLocalService.addObjectFolder(
+				RandomTestUtil.randomString(), TestPropsValues.getUserId(),
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				"Abl-e"));
+
+		AssertUtils.assertFailure(
+			DuplicateObjectFolderExternalReferenceCodeException.class,
```

Comment:
I believe this assert should come first, what do you think?

Reply:
Aggred.